rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -299,14 +305,19 @@ rosbag2_storage::SerializedBagMessageSharedPtr * Player::peek_next_message_from_
 
 bool Player::play_next()
 {
-  // Temporary take over playback from play_messages_from_queue()
-  std::lock_guard<std::mutex> lk(skip_message_in_main_play_loop_mutex_);
-
-  if (!clock_->is_paused() || !is_in_play_) {
+  if (!clock_->is_paused()) {
     return false;
   }
 
+  // Temporary take over playback from play_messages_from_queue()
+  std::lock_guard<std::mutex> main_play_loop_lk(skip_message_in_main_play_loop_mutex_);
   skip_message_in_main_play_loop_ = true;
+  // Wait for player to be ready for playback messages from queue i.e. wait for Player:play() to
+  // be called if not yet and queue to be filled with messages.
+  {
+    std::unique_lock<std::mutex> lk(ready_to_play_from_queue_mutex_);
```

Comment:
Will this block forever if `play` has not been called yet?

Reply:
Yes, it's going to wait forever while `play` will not be called. 
Otherwise it will be non deterministic behavior and race conditions will not be solved.
BTW. I will need the same synchronization for `Player::seek/jump` it's going to be the same problem with race conditions.

Reply:
@emersonknapp gentle ping to finish review.